cluster-autoscaler/core/scaledown/actuation/update_latency_tracker.go

```diff
@@ -0,0 +1,79 @@
+package actuation
+
+import (
+	"time"
+
+	"k8s.io/autoscaler/cluster-autoscaler/utils/kubernetes"
+	"k8s.io/autoscaler/cluster-autoscaler/utils/taints"
+	"k8s.io/klog/v2"
+)
+
+type UpdateLatencyTracker struct {
+	startTimestamp  map[string]time.Time
+	finishTimestamp map[string]time.Time
+	nodeLister      kubernetes.NodeLister
+}
+
+func NewUpdateLatencyTracker(startTimestamp map[string]time.Time, nodeLister kubernetes.NodeLister) *UpdateLatencyTracker {
+	return &UpdateLatencyTracker{
+		startTimestamp:  startTimestamp,
+		finishTimestamp: map[string]time.Time{},
+		nodeLister:      nodeLister,
+	}
+}
+
+func (u *UpdateLatencyTracker) Start(stopChan <-chan struct{}, resultChan chan int64) {
+	for {
+		select {
+		case <-stopChan:
+			var remainingNodes []string
+			for node, _ := range u.startTimestamp {
+				remainingNodes = append(remainingNodes, node)
+			}
+			for remainingNodes != nil {
+				time.Sleep(100 * time.Millisecond)
+				u.updateFinishTime()
+				remainingNodes = u.remainingNodesToFinish(remainingNodes)
+			}
+			latency := u.calculateLatency()
+			resultChan <- latency
+			return
+		default:
+		}
+		u.updateFinishTime()
+		time.Sleep(50 * time.Millisecond)
+	}
+}
+
+func (u *UpdateLatencyTracker) updateFinishTime() {
+	nodes, err := u.nodeLister.List()
```

Comment:
Actually, you don't need to list all nodes and iterate through them - you can iterate through nodes with start timestamp and call `Get` on the lister.

Reply:
Done